gtk4/src/expression.rs

```diff
@@ -0,0 +1,362 @@
+                property_name.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_property_expression_new_for_pspec")]
+    pub fn new_for_pspec<E: AsRef<Expression>>(
+        expression: Option<&E>,
+        pspec: glib::ParamSpec,
+    ) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_property_expression_new_for_pspec(
+                expression.map(|e| e.as_ref()).to_glib_none().0,
+                pspec.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_property_expression_get_expression")]
+    pub fn get_expression(&self) -> Expression {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_none(ffi::gtk_property_expression_get_expression(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_property_expression_get_pspec")]
+    pub fn get_pspec(&self) -> glib::ParamSpec {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_none(ffi::gtk_property_expression_get_pspec(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+}
+
+define_expression!(
+    ObjectExpression,
+    ffi::GtkObjectExpression,
+    ffi::gtk_object_expression_get_type
+);
+
+impl ObjectExpression {
+    #[doc(alias = "gtk_property_expression_new")]
+    pub fn new<T: IsA<Object>>(object: &T) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_object_expression_new(
+                object.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "gtk_object_expression_get_object")]
+    pub fn get_object(&self) -> Option<Object> {
+        skip_assert_initialized!();
+        unsafe { from_glib_none(ffi::gtk_object_expression_get_object(self.to_glib_none().0)) }
+    }
+}
+
+define_expression!(
+    ClosureExpression,
+    ffi::GtkClosureExpression,
+    ffi::gtk_closure_expression_get_type
+);
+
+impl ClosureExpression {
+    #[doc(alias = "gtk_closure_expression_new")]
+    pub fn new(value_type: Type, closure: glib::Closure, params: &[Expression]) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_closure_expression_new(
+                value_type.to_glib(),
+                closure.to_glib_none().0,
+                params.len() as u32,
+                params.to_glib_full(),
+            ))
+        }
+    }
+}
+
+define_expression!(
+    ConstantExpression,
+    ffi::GtkConstantExpression,
+    ffi::gtk_constant_expression_get_type
+);
+
+impl ConstantExpression {
+    #[doc(alias = "gtk_constant_expression_new")]
+    pub fn new<V: ToValue>(value: &V) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_constant_expression_new(
+                value.to_value_type().to_glib(),
+                [value.to_value().to_glib_none().0],
```

Comment:
This is wrong. It's a varargs function that collects a `GValue` from the following arguments. You'd call it e.g. like `gtk_constant_expression_new(G_TYPE_INT, 123, NULL)` and that would internally collect the `123` into a `GValue`.

Simply call `gtk_constant_expression_new_for_value()` here too

Reply:
Aha, right. Makes sense actually as the former's a very C api.